edb/edgeql/compiler/stmtctx.py

```diff
@@ -139,6 +140,10 @@ def fini_expression(
     ):
         ir = setgen.scoped_set(ir, ctx=ctx)
 
+    # Compile any triggers that were triggered by the query
+    ir_triggers = triggers.compile_triggers(
+        ast_visitor.find_children(ir, irast.MutatingStmt), ctx=ctx)
```

Comment:
Do we have any idea how slow those traversals are?  Any reason not to collect pending triggers in context as you make DML IR?

Reply:
I haven't measured it. I did a traversal because I think it gets hard to keep track of too much state, and we already have a `dml_stmts` that tracked something slightly different (deletes weren't included) and I didn't want to be tracking dml in two ways. I think the actually right answer though is to just track deletes in that path